src/Data/List/Properties.agda

```diff
@@ -317,18 +344,48 @@ module _ {f g : These A B → C} where
   map-alignWith g (x ∷ xs) (y ∷ ys) =
     cong₂ _∷_ refl (map-alignWith g xs ys)
 
+  alignWith-flip : ∀ xs ys →
+                 alignWith f xs ys ≡ alignWith (f ∘ These.swap) ys xs
+  alignWith-flip []       []       = refl
+  alignWith-flip []       (y ∷ ys) = refl
+  alignWith-flip (x ∷ xs) []       = refl
+  alignWith-flip (x ∷ xs) (y ∷ ys) = cong (_ ∷_) (alignWith-flip xs ys)
+
+module _ {f : These A A → B} where
+
+  alignWith-comm : f ∘ These.swap ≗ f →
+                 ∀ xs ys → alignWith f xs ys ≡ alignWith f ys xs
+  alignWith-comm f-comm xs ys = trans
+    (alignWith-flip xs ys)
+    (alignWith-cong f-comm ys xs)
+
+------------------------------------------------------------------------
+-- align
+
+align-map : ∀ (f : A → B) (g : C → D) →
+            ∀ xs ys → align (map f xs) (map g ys) ≡
+                      map (These.map f g) (align xs ys)
+align-map f g xs ys = trans
+  (alignWith-map f g xs ys)
+  (sym (map-alignWith (These.map f g) xs ys))
+
+module _ (xs : List A) (ys : List B) where
+
+  align-flip : align xs ys ≡ map These.swap (align ys xs)
+  align-flip = trans
+    (alignWith-flip xs ys)
+    (sym (map-alignWith These.swap ys xs))
+
 ------------------------------------------------------------------------
 -- zipWith
 
-module _ (f : A → A → B) where
+module _ {f g : A → B → C} where
```

Comment:
Eugh, I'm torn on this one whether this is a breaking change or not. As long as the user is not passing implicits explicitly it should be fine. Anyone else has have any feelings on it? 

If we decide it is fine, then we should definitely add a CHANGELOG entry for it.

Reply:
the diff is a little bit messy here, what happened is that [`zipWith-map` was added](https://github.com/agda/agda-stdlib/pull/2355/commits/76940a93cf9b1c36098f45988ca710a2731e1d53) (i put it at the start for consistency) and later [`zipWith-comm` was moved to the bottom](https://github.com/agda/agda-stdlib/pull/2355/commits/bb9a8a2c22266315175b2f61a36013a631e1f7e2) since it can be implemented more easily in terms of `zipWith-flip` and `zipWith-cong`, but there should be no breaking change

Reply:
Ah sorry, my misunderstanding!